src/clusterfuzz/_internal/bot/tasks/utasks/minimize_task.py

```diff
@@ -586,23 +573,66 @@ def utask_main(uworker_input):
     testcase.put()
 
     if check_deadline_exceeded_and_store_partial_minimized_testcase(
-        deadline, uworker_input.testcase_id, uworker_input.job_type,
-        input_directory, file_list, data, testcase_file_path):
-      return
+        deadline, testcase_id, uworker_input.job_type, input_directory,
+        file_list, data, testcase_file_path):
+      return None
 
   command = testcase_manager.get_command_line_for_application(
       testcase_file_path, app_args=app_arguments, needs_http=testcase.http_flag)
   last_crash_result = test_runner.last_failing_result
 
-  store_minimized_testcase(testcase, input_directory, file_list, data,
-                           testcase_file_path)
-  finalize_testcase(
-      uworker_input.testcase_id,
-      command,
-      last_crash_result,
+  return uworker_io.UworkerOutput(
+      testcase=testcase,
+      input_directory=input_directory,
+      file_list=file_list,
+      data=data,
```

Comment:
I think you also need to ensure any fields you have here are added to UworkerOutput in uworker_message.proto.
Since there's some minimize specific output fields here, I'd do what FuzzTaskOutput does instead of putting those fields directly on UworkerOutput.

Reply:
ahhhh yes that makes sense!!
I've added the crash results as a json to follow what FuzzTaskOutput does ish- unsure if its better to have that be a separate proto message though.